Source/santad/SNTEventLog.m

```diff
@@ -501,4 +508,52 @@ - (NSString *)serialForDevice:(NSString *)devPath {
   return serial;
 }
 
+/**
+ Uses the executable path, uid, and gid from in a given santa_message_t to determine if the path
+ has been translocated by GateKeeper and if so, returns the original path of the executable.  This
+ requires MacOS 10.12 or higher.  We use dlopen to access the functions we need in
+ Security.framework so that we can still build against the 10.11 SDK.  If the path has not been
+ translocated or if running on MacOS prior to 10.12, this method returns nil.
+ */
+- (NSString *)originalPathForTranslocation:(santa_message_t)message {
+  // The first time this function is called, we attempt to find the addresses of
+  // SecTranslocateIsTranslocatedURL and SecTranslocateCreateOriginalPathForURL inside of the
+  // Security.framework library.  If we were successful, handle will be non-NULL and is never
+  // closed.
+  static void *handle;
+  static Boolean (*isTranslocatedURL)(CFURLRef, bool *, CFErrorRef *);
+  static CFURLRef __nullable (*createOriginalPathForURL)(CFURLRef, CFErrorRef *);
```

Comment:
done. the static vars are default initialized to 0, but I think explicit initialization is better.